operator/pkg/helmreconciler/reconciler.go

```diff
@@ -308,8 +322,8 @@ func (h *HelmReconciler) addComponentLabels(coreLabels map[string]string, compon
 	for k, v := range coreLabels {
 		labels[k] = v
 	}
-	revision := ""
-	if h.iop != nil {
+	revision := coreLabels[label.IstioRev]
+	if revision == "" && h.iop != nil {
```

Comment:
--revision is an alias for --set revision, which is included in the final IOP once flags are applied. in that case i'm not sure why we need to pass revision around rather than getting it from IOP everywhere?

Reply:
because iop is not always non-nil? Or do you think even for the case in istioctl command it would be non-nil all the time?

Reply:
i'm still struggling with the case where iop is nil. when would that ever happen? if it's not nil, they we can use the revision field from that right?